odk1-src/form-audit-log.rst

```diff
@@ -0,0 +1,146 @@
++-------------------+------------------------------------------------------------------+-------+----------------+
+| add repeat        | Add a repeat                                                     | Yes   | Yes            |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| delete repeat     | Delete a repeat                                                  | Yes   | Yes            |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| end screen        | View the end screen                                              | No    | Yes            |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| form save         | Save the form                                                    | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| form exit         | Exit the form                                                    | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| form resume       | Resume the form                                                  | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| form finalize     | Finalize the form                                                | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| save error        | Error trying to save                                             | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| finalize error    | Error trying to finalize the form (probably encryption related)  | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| constraint error  | Constraint or required error on finalize                         | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+
+.. _audit-timestamps:
+
+Timestamps
+-----------
+
+Device time can be changed by the user unless Android access controls are put in place. To reduce the ways that audits can be manipulated by modifying device time, clock time is collected once at form launch and all following times are computed relative to that initial reading. This means that while the actual time may potentially be inaccurate, the time elapsed within and between screens is always accurate within one form editing session.
```

Comment:
This paragraph seems a little unclear. I read it a few times to grok it, because at first, it sounded like the audit stored elapsed time for each event. But it doesn't (right?) --- it calculates the presumed actual time *using* elapsed time (instead of re-checking the device clock).

I don't have a brilliant suggestion for improving it yet...

Reply:
Your interpretation is correct. Suggestions welcome and I'll try to play with it some more as well! The idea itself is tricky.

Reply:
What about this...

> If we relied entirely on the time reported by the device for every event, users (or the network) could change the device time and manipulate the correctness of the audit log. For this reason, only use device time for the form start timestamp. All subsequent event timestamps are the result of elapsed time, which users cannot change, added to the form start timestamp. This means that while the timestamps themselves may potentially be inaccurate, the time elapsed within and between the timestamps are always accurate within one form editing session.

Reply:
Thanks, @yanokwa, that is clearer!